Sources/NIOCore/EventLoop+SerialExecutor.swift

```diff
@@ -25,6 +25,9 @@ public protocol NIOSerialEventLoopExecutor: EventLoop, SerialExecutor { }
 extension NIOSerialEventLoopExecutor {
     @inlinable
     public func enqueue(_ job: consuming ExecutorJob) {
+        // By default we are just going to use execute to run the job
+        // this is quite heavy since it allocates the closure for
+        // every single job.
```

Comment:
can't we just `self.enqueue(job)` now?

Reply:
We can't do that because that would recursively call the method itself. I also tried removing this method here all together but that ends up in ambiguous look-up during compile time. So I think for now we just have to keep it here as well.